src/RiotAPILibraries.ts

```diff
@@ -0,0 +1,167 @@
+import { SharedSettings } from "./SharedSettings";
+import { PersonalSettings } from "./PersonalSettings";
+import fetch from "node-fetch";
+
+import Discord = require("discord.js");
+
+
+interface LinkStruct {
+    self: string;
+    git: string;
+    html: string;
+}
+
+interface GithubAPIStruct {
+    name: string;
+    path: string;
+    sha: string;
+    size: number;
+    url: string;
+    html_url: string;
+    git_url: string;
+    download_url: string;
+    type: string;
+    _links: LinkStruct;
+}
+
+interface APILibraryLink {
+    name: string,
+    url: string,
+}
+
+interface APILibraryStruct {
+    owner: string;
+    repo: string;
+    language: string;
+    description: string;
+    links: APILibraryLink[];
+    metadata: any;
+    tags: string[];
+}
+
+interface LibraryDescription
+{
+    valid: boolean,
+    stars: number,
+    library: APILibraryStruct | null,
+    links: string[]
+}
+
+export default class RiotAPILibraries {
+    private bot: Discord.Client;
+    private settings: SharedSettings;
+
+    private lastCall: number;
+
+    private fetchSettings: object;
+
+    constructor(bot: Discord.Client, personalSettings: PersonalSettings, settings: SharedSettings) {
+        console.log("Requested Github extension..");
+        this.bot = bot;
+        this.settings = settings;
+
+        this.bot.on("ready", this.onBot.bind(this));
+        this.bot.on("message", this.onCommand.bind(this));
+
+        this.fetchSettings = {
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "application/json",
+                Authorization: `Basic ${new Buffer(personalSettings.github.username + ":" + personalSettings.github.password, "binary").toString("base64")}`
```

Comment:
`new Buffer` is deprecated. Use `Buffer.from(contents)` instead. No need to specify encoding, it is utf8 by default.